### PR TITLE
Feat/text input updated attributes

### PR DESCRIPTION
**Pull request by jamesducky**
Closes [UpdateTextInput: Add input types and validation](https://app.asana.com/0/1198919072280397/1202559938332884/f)

### Summary of changes included in this PR

- Adds constraint validation attributes: min, max, minLength, maxLength, step, pattern
- Adds input attributes: autocomplete, width, name, placeholder, type(email, search, number, tel, URL, tel, email and password)
- Adds the [validityState object](https://developer.mozilla.org/en-US/docs/Web/API/ValidityState) to the event emitter to enable customised error message
- Updates stories
- Updates unit test
- Updates react example

### Reviewer checklist:

- Tests for
  - accessibility
  - rendered HTML (spec)
- Storybook stories for all variants
- JSDoc documentation for component
- Update Example app (React, ...)

### CODE REVIEW

File: packages/component-library/src/components/plmg-text-input/plmg-text-input.stories.js [hunk 1]
```diff
@@ -66,30 +99,26 @@ const Template = (args) => {
...
-    .trim();
-
-  const el = document.createElement('div');
-  el.innerHTML = htmlContent;
-  el.style.display = 'flex';
-  el.style.justifyContent = 'space-between';
-
-  return el;
+Primary.argTypes = {
+  ['min']: { control: { disable: true } },
```
Comment (TanguyDucky) [inline packages/component-library/src/components/plmg-text-input/plmg-text-input.stories.js:118]: Why disabling them? It prevents me from testing the text-input of type number :confused:

File: packages/component-library/src/components/plmg-text-input/plmg-text-input.tsx [hunk 2]
```diff
@@ -18,6 +21,25 @@ import {
...
+  @Element()
+  el: HTMLElement;
+  /**
+   * Define autocomplete
+   *
+   * Allowed value: any string
+   *
+   * Default: off
+   */
+  @Prop() autoComplete: string = 'off';
```
Comment (TanguyDucky) [inline packages/component-library/src/components/plmg-text-input/plmg-text-input.tsx:36]:
```suggestion
  @Prop() autoComplete: 'on' | 'off' = 'off';
```

File: packages/component-library/src/components/plmg-text-input/plmg-text-input.tsx [hunk 3]
```diff
@@ -44,48 +67,149 @@ export class TextInput {
...
+   *
+   * Default: none
    */
-  @Prop() showLabel: boolean = true;
-  @Watch('showLabel')
-  validateShowLabel(newValue: boolean) {
-    if (newValue && typeof newValue !== 'boolean')
-      throw new Error('show label must be boolean');
+  @Prop() max: number;
+  @Watch('maxLength')
```
Comment (TanguyDucky) [inline packages/component-library/src/components/plmg-text-input/plmg-text-input.tsx:97]:
```suggestion
  @Watch('max')
```

File: packages/component-library/src/components/plmg-text-input/plmg-text-input.tsx [hunk 4]
```diff
@@ -44,48 +67,149 @@ export class TextInput {
...
+  /**
+   * Define the min value when type is number
+   *
+   * Allowed values:
+   * - Any number
+   *
+   * Only vaiid for type: number
+   */
+  @Prop() min: number;
+  @Watch('minLength')
```
Comment (TanguyDucky) [inline packages/component-library/src/components/plmg-text-input/plmg-text-input.tsx:128]:
```suggestion
  @Watch('min')
```

File: packages/component-library/src/components/plmg-text-input/plmg-text-input.tsx [hunk 5]
```diff
@@ -195,17 +408,50 @@ export class TextInput {
...
-            value={this.internalValue}
-            onInput={(ev) => this.handleInputChange(ev)}
-          />
+          {this.type === 'number' ? (
+            <input
+              autoComplete={this.autoComplete}
+              class={inputClasses}
+              disabled={this.disabled}
+              id={this.labelToId()}
+              max={this.max}
```
Comment (TanguyDucky) [inline packages/component-library/src/components/plmg-text-input/plmg-text-input.tsx:417]: I'm not sure you need to separate the 2 cases. Have you try providing all the attributes regardless of the type?
I'm pretty sure a input of type number will ignore `maxLength` for example.
And an input of type password will ignore `max` and `min`.

Comment (TanguyDucky) [reply packages/component-library/src/components/plmg-text-input/plmg-text-input.tsx:417]: and it leads to duplication problem, for example you forgot to duplicate the `width` logic.

File: packages/component-library/src/components/plmg-text-input/plmg-text-input.tsx [hunk 6]
```diff
@@ -195,17 +408,50 @@ export class TextInput {
...
+              min={this.min}
+              name={this.name}
+              onInput={(event) => this.handleInputChange(event.target)}
+              placeholder={this.placeholder}
+              readonly={this.readOnly}
+              required={this.required}
+              size={this.width}
+              step={this.step}
+              style={{
```
Comment (TanguyDucky) [inline packages/component-library/src/components/plmg-text-input/plmg-text-input.tsx:426]: missing from the other case

File: packages/component-library/src/components/plmg-text-input/plmg-text-input.stories.js [hunk 7]
```diff
@@ -1,56 +1,89 @@
...
+    },
+    ['read-only']: { control: { type: 'boolean' } },
+    required: {
+      control: { type: 'boolean' },
+    },
     tip: {
       control: { type: 'text' },
     },
+    type: {
+      options: ['text', 'password', 'email', 'number', 'tel', 'url'],
```
Comment (TanguyDucky) [inline packages/component-library/src/components/plmg-text-input/plmg-text-input.stories.js:55]:
```suggestion
      options: ['text', 'password', 'email', 'number', 'search', 'tel', 'url'],
```

File: packages/component-library/src/components/plmg-text-input/plmg-text-input.tsx (hunk 5 above)
Comment (jamesducky) [reply packages/component-library/src/components/plmg-text-input/plmg-text-input.tsx]: Yes. Works fine. Tests show that indeed irrelevant props are ignored. I was using the 'size' attribute to set the width in preference to inline styling, but that has no effect on the number.

File: packages/component-library/src/components/plmg-text-input/plmg-text-input.tsx (hunk 6 above)
Comment (jamesducky) [reply packages/component-library/src/components/plmg-text-input/plmg-text-input.tsx]: Explanation above. I was using size to set width with non-number fields

**Pull request merged**